lib/src/reddit.dart

```diff
@@ -125,6 +128,75 @@ class Reddit {
     throw new DRAWAuthenticationError('Unable to authenticate with Reddit');
   }
 
+  /// Creates a new authenticated [Reddit] instance from cached credentials.
+  ///
+  /// [credentialsJson] is a JSON string containing the cached credentials. This
+  /// parameter is required.
+  ///
+  /// This string can be retrieved from an authenticated [Reddit] instance in
+  /// the following manner:
+  ///
+  /// ```dart
+  /// final credentialsJson = reddit.auth.credentials.toJson();
+  /// ```
+  ///
+  /// [clientId] is the identifier associated with your authorized application
+  /// on Reddit. To get a client ID, create an authorized application
+  /// [here](http://www.reddit.com/prefs/apps).
+  ///
+  /// [clientSecret] is the unique secret associated with your client ID. This
+  /// is required for script and web applications.
+  ///
+  /// [userAgent] is an arbitrary identifier used by the Reddit API to
+  /// differentiate between client instances. This should be relatively unique.
+  ///
+  /// [redirectUri] is the redirect URI associated with your Reddit application.
+  /// This field is unused for script and read-only instances.
+  ///
+  /// [tokenEndpoint] is a [Uri] to an alternative token endpoint. If not
+  /// provided, [defaultTokenEndpoint] is used.
+  ///
+  /// [authEndpoint] is a [Uri] to an alternative authentication endpoint. If not
+  /// provided, [defaultAuthTokenEndpoint] is used.
+  ///
+  /// [configUri] is a [Uri] pointing to a 'draw.ini' file, which can be used to
+  /// populate the previously described parameters.
+  ///
+  /// [siteName] is the name of the configuration to use from draw.ini. Defaults
+  /// to 'default'.
+  static Future<Reddit> restoreAuthenticatedInstance(
+      {String clientId,
+      String clientSecret,
+      String userAgent,
+      String credentialsJson,
+      Uri redirectUri,
+      Uri tokenEndpoint,
+      Uri authEndpoint,
+      Uri configUri,
+      String siteName = 'default'}) async {
+    if (credentialsJson == null) {
+      throw new DRAWArgumentError('credentialsJson cannot be null.');
```

Comment:
Should we enforce the `credentialsJson` to be a required parameter in the function contract?
i.e
```dart
static Future<Reddit> restoreAuthenticatedInstance(String credentialsJson,
{String clientId,
       String clientSecret,
       String userAgent,
       String credentialsJson,
       Uri redirectUri,
       Uri tokenEndpoint,
       Uri authEndpoint,
       Uri configUri,
       String siteName = 'default'}) async {
```

Reply:
Yeah, it might be better to do it your way. It's a shame that there's no good way to have required named parameters :(